IntegrationTests/IntegrationScoped.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace IntegrationTests
+{
+    public class IntegrationScoped<TStrategy> where TStrategy : IScopedStrategy, new()
+    {
+
+        protected readonly TStrategy Strategy = new TStrategy();
+
+        [TestInitialize]
+        public void Setup()
+        {
+            Strategy.Setup().Wait();
```

Comment:
You still need a setup and teardown. At least you can parameterize it. I am still thinking about how to make the setup / teardown fully async. in the works.